### PR TITLE
Get Mime type without fileinfo extension.

### PR DESCRIPTION
**Pull request by jabedbd**
Some time it is not able to install or enable the fileinfo extension in shared hosting. This change able to run flarum without fileinfo extension.

### CODE REVIEW

**Comment by franzliedke** [comment]
Hmm, wouldn't that just mean we now require the exif extension instead?

**Comment by jabedbd** [comment]
exif extension mostly installed by default with shared hosting server.

**Comment by franzliedke** [comment]
Are there any stats about enabled extensions on various hosting providers?

**Comment by jabedbd** [comment]
I didn't found any stats about installation of fileinfo and exif. Everything i said from my little experiences as a poor shared hosting user :p 
My second source of research is stackoverflow. I found a lot of issues there with the fileinfo and shared hosting where issues about exif is less (almost zero) :D

**Comment by AndreiTelteu** [comment]
Would be better to check if those functions exists ?
```php
$mimetype = pathinfo($file, PATHINFO_EXTENSION) === 'svg'
    ? 'image/svg+xml'
    : (function_exists('finfo_file')
        ? finfo_file(finfo_open(FILEINFO_MIME_TYPE), $file)
        : image_type_to_mime_type(exif_imagetype($file)));
```
Not this ugly, maybe do a separate method for this.

**Comment by luceos** [comment]
I tried researching the availability of both extensions, but found nothing conclusive. Thus I'd say let's not change anything and possibly introduce unforeseen issues.

I do like the idea of @AndreiTelteu , but then we're checking against several extensions without any fixed dependency, thus also not having the certainty of a fallback.

tldr, I'm saying no to this.

**Comment by tobscure** [comment]
What about a simple switch/case for the file extension? (eg. svg → image/svg+xml, png → image/png, etc)

There's a chance we will refactor this code in the future to actually refer to the image file rather than a `data:` URI, but I'm still unsure about that so happy to discuss a solution to reduce dependency on certain PHP extensions.

**Comment by luceos** [comment]
Why are we even writing a custom method to identify the mimetype. Looking at Symfony:

```
The mime type is guessed using a MimeTypeGuesser instance, which uses finfo(), mimecontenttype() and the system binary "file" (in this order), depending on which of those are available.
```

**Comment by luceos** [comment]
@franzliedke can add your opinion, it feels like we spent too much time on this already at the cost of @jabedbd

**Comment by franzliedke** [comment]
I agree: if we manage to get rid of requiring fileinfo (without replacement with another extension), then I'd be happy to accept such a PR. Toby's suggestion seems simple enough.

@jabedbd Would you like to implement this feature?

**Comment by jabedbd** [comment]
@franzliedke  yes,i like to do it. let me try.

**Comment by luceos** [comment]
Why add a hack switch statement when we can use symfony which is already part of flarum anyway @franzliedke ?

**Comment by franzliedke** [comment]
@luceos If I understood your earlier post correctly, Symfony only falls back to the `file` binary, which isn't exactly what we are doing here (but which should be just as fine, right)?

**Comment by franzliedke** [comment]
@jabedbd Can you push that change (to a simple `switch` statement) or do you want us to do it?

**Comment by jabedbd** [comment]
@franzliedke i'm busy with a project. please do it.

**Comment by franzliedke** [comment]
Done in https://github.com/flarum/core/commit/ba96f311a960314104becc716cd3dc7b5291cc6d.

Wasn't there an issue about our reliance on certain PHP extensions over others. I can't seem to find it, not sure why...
